### PR TITLE
SAM-2959; null pointer if no honor pledge value

### PR DESCRIPTION
**Pull request by clhedrick**

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/delivery/LoginServlet.java [hunk 1]
```diff
@@ -135,7 +135,10 @@ public void doPost(HttpServletRequest req, HttpServletResponse res)
     }
     delivery.setAssessmentId(pub.getPublishedAssessmentId().toString());
     delivery.setAssessmentTitle(pub.getTitle());
-    delivery.setHonorPledge(pub.getAssessmentAccessControl().getHonorPledge());
+    Boolean honorPledge = pub.getAssessmentAccessControl().getHonorPledge();
+    if (honorPledge == null)
+	honorPledge = false;
```
Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/delivery/LoginServlet.java]: Please use braces on single line conditionals.

Comment (relong76) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/delivery/LoginServlet.java]: boolean honorPledge = BooleanUtils.toBooleanDefaultIfNull(pub.getAssessmentAccessControl().getHonorPledge(), false);

would be cleaner. (And who doesn't love Apache Commons?) :)

Comment (bjones86) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/delivery/LoginServlet.java]: Apache Commons to the rescue, yet again!!

**Comment by clhedrick** [comment]
OK, I'm using booleanutils, though a simpler method from it.

**Comment by bjones86** [comment]
LGTM

**Comment by bjones86** [comment]
Please use the squash and merge feature....

**Comment by clhedrick** [comment]
I thought I had. The button claimed it was.

On Jun 30, 2016, at 12:53 PM, Brian Jones <notifications@github.com<mailto:notifications@github.com>> wrote:


Please use the squash and merge feature....

**Comment by bjones86** [comment]
Oh sorry Chuck, my mistake, you are correct. I just didn't look closely enough. My apologies!

**Comment by clhedrick** [comment]
That’s actually not the result I had expected. It gave all the separate commit messages. This is the first time I had used it. I was hoping it would let me supply a single commit message, like squash does.

On Jun 30, 2016, at 1:00 PM, Brian Jones <notifications@github.com<mailto:notifications@github.com>> wrote:


Oh sorry Chuck, my mistake, you are correct. I just didn't look closely enough. My apologies!

**Comment by bjones86** [comment]
It does actually give you that opportunity. When you select the option from the drop down, a text box appears underneath with all the different commit messages dumped into the extra info message, with the actual commit message above in another text box. This gives you the opportunity to both amend the commit message along with all the extra (text) details.

**Pull request merged**